### PR TITLE
Updating to use AWS services

### PR DESCRIPTION
**Pull request by helrond**
Rearchitects this site to use AWS Lambda and S3:
- GitHub webhooks trigger a Lambda task
- Lamda pulls repositories, builds and pushes relevant site to S3, where it can be served up

This means one less server for us to manage locally, and build processes that are event-driven (rather than scheduled at a particular time of day).

I also made some changes to the Dockerfile used for local development to simplify what's going on there.

Casting a wide net with reviews here as a way of informing about this change but also because I want to make sure the documentation is understandable for someone who doesn't have intimate knowledge of AWS services. I am happy to run a walkthrough session with folks if that would be useful - if so, drop a comment below!

### CODE REVIEW

**Comment by helrond** [comment]
I should note that one effect of this is that we'll have to pull Aurora and Argo docs out into their own repos (rather than being in `docs-*` branches in those main repositories.

File: README.md [hunk 1]
```diff
@@ -1,96 +1,80 @@
 # docs-build
 
 Theme file along with a Python script which retrieves updates for documentation
-repositories, and builds internal and external-facing sites. It can be set to
-run at a specified time (for example on a nightly basis) using cron. Requires at
+repositories, and builds internal and external-facing sites. Requires at
 least one documentation repository (see [processing-manual](https://github.com/RockefellerArchiveCenter/processing-manual)).
 
 ![Build Status](https://travis-ci.org/RockefellerArchiveCenter/docs-build.svg?branch=base)](https://travis-ci.org/RockefellerArchiveCenter/docs-build)
```
Comment (HaSistrunk) [inline README.md]:
```suggestion
[![Build Status](https://app.travis-ci.com/RockefellerArchiveCenter/docs-build.svg?branch=base)](https://app.travis-ci.com/RockefellerArchiveCenter/docs-build)
```

**Comment by HaSistrunk** [comment]
This is exiting! I'm still checking this out to make sure I understand the infrastructure, and I would be interested in a quick demo just to see what things look like in AWS to better understand the full picture.

I have a few questions just to make sure I understand how the workflows will change for updating/adding new docs.
1. Does this do away with the Dev site?
2. To implement, we'll have to add webhooks to all the current docs repos, right? I'm not clear what that should look like.
3. If I'm looking at `repositories.yml` correctly, it may be useful to mention in the README that private docs get added to both the public and private sections of `repositories.yml`. This logic makes sense to me, but wasn't immediately clear.

**Comment by helrond** [comment]
@HaSistrunk Cool, I can look at scheduling a walkthrough of all of this soon-ish. Would be good to know if other folks are also interested.

To answer your questions:
1. nope, dev site is still in place, but we'll just build the private site in dev since it's only going to be available internally anyway.
2. I have already added these webhooks on all the docs repos, so you can poke around and see what they look like. I've also linked to the main GH Webhooks documentation from the README. If there's any additional info you'd like to see in the README that would be good to know.
3. Thanks - updated!

**Comment by p-galligan** [comment]
> @HaSistrunk Cool, I can look at scheduling a walkthrough of all of this soon-ish. Would be good to know if other folks are also interested.

I'm definitely interested in a walkthrough and seeing how this works.

**Comment by daremiyo** [comment]
> @HaSistrunk Cool, I can look at scheduling a walkthrough of all of this soon-ish. Would be good to know if other folks are also interested.
> 

I'd also appreciate a demo! It would be really helpful for better understanding the changes

**Comment by kemartin2015** [comment]
I would also be interested in attending a walkthrough session.

**Comment by HaSistrunk** [comment]
This is looking good to me! We can remove the `.gitmodules`, `.gitmodules-dev` and the `/repositories` directory now that we aren't depending on gitmodules, right?

**Review by p-galligan — APPROVED** [review]

**Review by kemartin2015 — APPROVED** [review]

**Review by HaSistrunk — APPROVED** [review]

**Review by daremiyo — APPROVED** [review]

**Review by bonniegee — APPROVED** [review]

**Pull request merged**